### PR TITLE
[clang] Introduce `SemaRISCV`

### PR DESCRIPTION
**Pull request by Endilll**
This patch moves `Sema` functions that are specific for RISC-V into the new `SemaRISCV` class. This continues previous efforts to split `Sema` up. Additional context can be found in https://github.com/llvm/llvm-project/pull/84184.

This PR is somewhat different from previous PRs on this topic:
1. Splitting out target-specific functions wasn't previously discussed. It felt quite natural to do, though.
2. I had to make some static function in `SemaChecking.cpp` member functions of `Sema` in order to use them in `SemaRISCV`.
3. I dropped "RISCV" from identifiers, but decided to leave "RVV" (RISC-V "V" vector extensions) intact. I think it's an idiomatic abbreviation at this point, but I'm open to input from contributors in that area.
4. I repurposed `SemaRISCVVectorLookup.cpp` for `SemaRISCV`.

I think this was a successful experiment, which both helps the goal of splitting `Sema` up, and shows a way to approach `SemaChecking.cpp`, which I wasn't sure how to approach before. As we move more target-specific function out of there, we'll gradually make the checking "framework" inside `SemaChecking.cpp` public, which is currently a whole bunch of static functions. This would enable us to move more functions outside of `SemaChecking.cpp`.

### CODE REVIEW

**Comment by wangpc-pp** [comment]
Is it a NFC?

**Comment by Endilll** [comment]
It is, but I guess significant enough to not be disguised by labeling as NFC.

**Comment by wangpc-pp** [comment]
I think it's a good rewrite. Added more RISCV guys.

**Review by cor3ntin — APPROVED** [review]

**Review by kito-cheng — APPROVED** [review]
LGTM as the original author of `SemaRISCVVectorLookup.cpp` :)

It's great to see we can put all RISC-V related stuff within same place rather than many different files.

**Comment by MaskRay** [comment]
Nice refactoring! Pushing a prerequisite commit first with `git mv SemaRISCVVectorLookup.cpp SemaRISCV.cpp` should make git recognize this as a file rename. This PR can be changed to rebase on that precommit.

**Comment by Endilll** [comment]
> Pushing a prerequisite commit first with git mv SemaRISCVVectorLookup.cpp SemaRISCV.cpp should make git recognize this as a file rename. This PR can be changed to rebase on that precommit.

Thank you for suggestion! Done.

File: clang/lib/Sema/SemaRISCV.cpp [hunk 1]
```diff
@@ -1,12 +1,12 @@
-//==- SemaRISCVVectorLookup.cpp - Name Lookup for RISC-V Vector Intrinsic -==//
+//==------- SemaRISCV.cpp ------- RISC-V target-specific routines ----------==//
```
Comment (MaskRay) [inline clang/lib/Sema/SemaRISCV.cpp:1]: The prevailing style starts with `//===-` or `//===--`

**Review by MaskRay — APPROVED** [review]

File: clang/lib/Sema/SemaRISCV.cpp (hunk 1 above)
Comment (Endilll) [reply clang/lib/Sema/SemaRISCV.cpp]: I never noticed this. Thanks!

**Review by wangpc-pp — APPROVED** [review]
LGTM.

File: clang/include/clang/Sema/SemaRISCV.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
+//===----- SemaRISCV.h ------- RISC-V target-specific routines ------------===//
```
Comment (topperc) [inline clang/include/clang/Sema/SemaRISCV.h:1]: I think header files are supposed to have `-*- C++ -*-` on this line

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**